NDTensors/src/diag/diagtensor.jl

```diff
@@ -145,16 +156,14 @@ function permutedims!(
   f::Function=(r, t) -> t,
 ) where {N}
   # TODO: check that inds(R)==permute(inds(T),perm)?
-  for i in 1:diaglength(R)
-    @inbounds setdiagindex!(R, f(getdiagindex(R, i), getdiagindex(T, i)), i)
-  end
+  data(R) .= f.(diagview(R), diagview(T))
```

Comment:
```suggestion
  diagview(R) .= f.(diagview(R), diagview(T))
```

Reply:
For some reason the first time I wrote that as `diagview` the code was not working properly. But now it is, perhaps some old bit of code was in my repl when I was testing. Updated now, thanks!